### PR TITLE
Fix typing behaviour on PID inputs

### PR DESCRIPTION
**Pull request by Sthirs**
Setting the PID input value while calculating the slider display
was the cause of losing the caret position.

refs #2009

### CODE REVIEW

File: src/js/TuningSliders.js [hunk 1]
```diff
@@ -253,36 +253,26 @@ TuningSliders.legacyUpdateFilterSlidersDisplay = function() {
...
     }
 
-    FC.PID_NAMES.forEach(function(elementPid, indexPid) {
+    FC.PID_NAMES.forEach(function (elementPid, indexPid) {
         const pidElements = $(`.pid_tuning .${elementPid} input`);
         pidElements.each(function (indexInput) {
-            if (indexPid < 3 && indexInput < 5) {
-                if (currentPIDs[indexPid * 5 + indexInput] != $(this).val()) {
+            if (indexPid < 3 && indexInput < 3) {
+                if ($(this).val() != FC.PIDS[indexPid][indexInput]) {
```
Comment (haslinghuis) [inline src/js/TuningSliders.js:273]:
```suggestion
                if ($(this).val() !== FC.PIDS[indexPid][indexInput]) {
```

**Review by haslinghuis — COMMENTED** [review]
Thanks!

**Review by haslinghuis — APPROVED** [review]
Did not test but LGTM.

**Comment by haslinghuis** [comment]
Testing: 

```
TuningSliders.js:273 Uncaught TypeError: Cannot read property '0' of undefined
    at HTMLInputElement.<anonymous> (TuningSliders.js:273)
    at Function.each (jquery.min.js:2)
    at S.fn.init.each (jquery.min.js:2)
    at TuningSliders.js:271
    at Array.forEach (<anonymous>)
    at Object.TuningSliders.updatePidSlidersDisplay (TuningSliders.js:269)
    at HTMLInputElement.<anonymous> (pid_tuning.js:592)
    at HTMLInputElement.dispatch (jquery.min.js:2)
    at HTMLInputElement.v.handle (jquery.min.js:2)
    at Object.trigger (jquery.min.js:2)
    at HTMLInputElement.<anonymous> (jquery.min.js:2)
    at Function.each (jquery.min.js:2)
    at S.fn.init.each (jquery.min.js:2)
    at S.fn.init.trigger (jquery.min.js:2)
    at S.fn.init.S.fn.<computed> [as change] (jquery.min.js:2)
    at pid_and_rc_to_form (pid_tuning.js:595)
```

**Comment by Sthirs** [comment]
Unfortunately I cannot reproduce the error. Could you please provide more info?

**Comment by haslinghuis** [comment]
To reproduce: type defaults in cli and go to pid tuning tab will show the error for me.

[diff.txt](https://github.com/betaflight/betaflight-configurator/files/6412342/diff.txt)

**Comment by Zuldan** [comment]
Currently in Betaflight Configurator v4.3 Nightly (v20210413.2497), if you have the sliders enabled and manually change the PID values then click on “Save”. The current slider positions and their associated values will override the PID values you manually entered in. I’m not sure this is the expected behaviour. I think the sliders should automatically be disabled when you manually enter in PID values.

**Comment by haslinghuis** [comment]
@Zuldan firmware version 4.3 introduces firmware sliders while 4.1 and 4.2 use legacy sliders.
After setting firmware sliders hit save. From here set the sliders to OFF to tune PID values manually (needs save again).
Agree this needs some polishing to make it more transparent.

**Comment by Sthirs** [comment]
@haslinghuis  I have tried with 3 different FCs and different versions of FW but I still can't reproduce the error.
Looking at the code it seems that `FC.PIDS` is `null`.
Do you have the same behaviour in the master too?
Could you please try putting this `console.log(FC.PIDS);` [here](https://github.com/betaflight/betaflight-configurator/blob/debf25adbd9997af39210115036dbd5320e5e6ee/src/js/TuningSliders.js#L262)?

**Comment by haslinghuis** [comment]
@Sthirs :confused:  Now I can't reproduce anymore. But indeed it could have been a timeout on the MSP call. Working on serial occasional the callback is invoked before the data is there.

Added and removed the console.log and it shows:

```
(10) [Array(3), Array(3), Array(3), Array(3), Array(3), Array(3), Array(3), Array(3), Array(3), Array(3), __ob__: Observer]
0: (3) [42, 85, 35]
1: (3) [46, 90, 38]
2: (3) [45, 90, 0]
3: (3) [50, 50, 75]
4: (3) [40, 0, 0]
5: (3) [undefined, undefined, undefined]
6: (3) [undefined, undefined, undefined]
7: (3) [undefined, undefined, undefined]
8: (3) [undefined, undefined, undefined]
9: (3) [undefined, undefined, undefined]
length: 10
```

**Comment by Sthirs** [comment]
So it might not be caused by the PR right?
In any case we could put something like this at the beginning of the function
```
if (FC.PIDS === null) {
    this.pidSlidersUnavailable = true;
}
```
but I don't think it's the right approach, it mitigates the problem and doesn't solve it, what do you think?

**Comment by haslinghuis** [comment]
Yes agree, better to let the error happen so we have an opportunity to look at the root cause.

**Review by mikeller — APPROVED** [review]

**Pull request merged**